controller/tests/controller_server/csi_controller_server_test.py

```diff
@@ -1264,6 +1260,132 @@ def test_controller_get_capabilities(self):
         self.servicer.ControllerGetCapabilities(request, context)
 
 
+class TestControllerServerExpandVolume(AbstractControllerTest):
+
+    def get_create_object_method(self):
+        return self.servicer.ControllerExpandVolume
+
+    def get_create_object_response_method(self):
+        return csi_pb2.ControllerExpandVolumeResponse
+
+    def setUp(self):
+        super().setUp()
+        self.fqdn = "fqdn"
+        self.mediator = XIVArrayMediator("user", "password", self.fqdn)
+        self.mediator.client = Mock()
+
+        self.storage_agent = MagicMock()
+        self.storage_agent.get_mediator.return_value.__enter__.return_value = self.mediator
+
+        self.servicer = ControllerServicer(self.fqdn)
+
+        self.access_types = csi_pb2.VolumeCapability.AccessMode
+        self.fs_type = "ext4"
+
+        self.mediator.maximal_volume_size_in_bytes = 3
+        self.mediator.minimal_volume_size_in_bytes = 1
+
+        self.request.volume_capability = csi_pb2.VolumeCapability(
+            access_mode=csi_pb2.VolumeCapability.AccessMode(mode=self.access_types.SINGLE_NODE_WRITER),
+            mount=csi_pb2.VolumeCapability.MountVolume(fs_type=self.fs_type))
+        self.request.parameters = {}
+        self.capacity_bytes = 2
+        self.request.capacity_range = Mock()
+        self.request.capacity_range.required_bytes = self.capacity_bytes
+        self.volume_id = "vol-id"
+        self.request.volume_id = "{}:{}".format("xiv", self.volume_id)
+        self.request.volume_content_source = None
+        self.context = utils.FakeContext()
+
+    def _prepare_expand_volume_mocks(self, storage_agent):
+        storage_agent.return_value = self.storage_agent
+
+        self.mediator.expand_volume = Mock()
+        self.mediator.expand_volume.return_value = utils.get_mock_mediator_response_volume(self.capacity_bytes,
+                                                                                           volume_name,
+                                                                                           self.volume_id,
+                                                                                           "a9k")
+
+    @patch("controller.controller_server.csi_controller_server.get_agent")
+    def test_expand_volume_with_size_too_small_fail(self, storage_agent):
+        self._prepare_expand_volume_mocks(storage_agent)
+
+        self.request.capacity_range.required_bytes = 0

```

Comment:
`A value of 0 is equal to an unspecified field value.`
I think we should return OK without expanding in this case.
do you?

Reply:
Its only in create, I didnt saw that in expand too

Reply:
ok, so do we interpret that `required_bytes` is mandatory for expand just because `CapacityRange` is `REQUIRED` there?
if so, does that mean we should interpret that `limit_bytes` is also mandatory?

Reply:
good question, I ignored it like done in create, its not very informative.

Reply:
1. do you think it is correct for create to ignore it?
2. if we align expand and create behavior regarding `limit_bytes`, why shouldn't we do the same for `required_bytes` (i.e. `A value of 0 is equal to an unspecified field value.`)?

Reply:
done